core/src/test/java/org/apache/iceberg/util/TestTableScanUtil.java

```diff
@@ -136,6 +141,129 @@ public void testTaskMerging() {
     Assert.assertEquals("Appropriate tasks should be merged", 3, mergedTasks.size());
   }
 
+  private static final Schema TEST_SCHEMA =
+      new Schema(
+          Types.NestedField.optional(1, "c1", Types.IntegerType.get()),
+          Types.NestedField.optional(2, "c2", Types.StringType.get()),
+          Types.NestedField.optional(3, "c3", Types.StringType.get()),
+          Types.NestedField.optional(4, "c4", Types.StringType.get()));
+
+  private static final PartitionSpec SPEC1 =
+      PartitionSpec.builderFor(TEST_SCHEMA).identity("c1").identity("c2").build();
+
+  private static final PartitionSpec SPEC2 =
+      PartitionSpec.builderFor(TEST_SCHEMA).identity("c1").identity("c3").identity("c2").build();
+
+  private static final StructLike PARTITION1 = new TestStructLike(100, "a");
+  private static final StructLike PARTITION2 = new TestStructLike(200, "b");
+
+  @Test
+  public void testTaskGroupPlanningByPartition() {
```

Comment:
Can we add a test that we throw an exception if the combining key includes columns not present in all partition specs? Like if we try to combine by all partition fields in `SPEC2`?

I think that should be covered by the existing code as `StructProjection$create` will complain.

Reply:
Thanks, added